### PR TITLE
Add Actor.Leader() method. Returns the Peer ID of the leader.

### PR DESCRIPTION
**Pull request by hsanjuan**
This is probably handy

### CODE REVIEW

File: actor.go [hunk 1]
```diff
@@ -69,3 +69,8 @@ func (actor *Actor) IsLeader() bool {
 	}
 	return false
 }
+
+// Leader returns the Raft's leader as seen by Raft (peer ID)
+func (actor *Actor) Leader() string {
```
Comment (whyrusleeping) [inline actor.go]: Maybe the return type should be `peer.ID`?

Comment (hsanjuan) [inline actor.go]: Yeah, I kind of agree. Added a new commit for it.

File: actor.go [hunk 2]
```diff
@@ -69,3 +69,14 @@ func (actor *Actor) IsLeader() bool {
...
 	return false
 }
+
+// Leader returns the LibP2P ID of the Raft leader.
+func (actor *Actor) Leader() peer.ID {
+	// Leader as returned by Libp2pTransport.LocalAddr()
+	raftLeader := actor.Raft.Leader()
+	peerID, err := peer.IDB58Decode(raftLeader)
+	if err != nil {
+		panic(err)
```
Comment (whyrusleeping) [inline actor.go]: Sketchy to panic, but alright. Is there no possible situation we could end up with the leader reporting an invalid peer ID as their name?

Comment (hsanjuan) [inline actor.go]: I had that same feeling. That string comes from the `LocalAddr()` in transport.go in the end, which does a `peer.IDB58Encode(t.host.ID())`. If this is invalid here something is really badly broken somewhere else. I'm happy to blow up the whole thing in that case.

**Pull request merged**